### ISSUE TITLE
Should isolation be added conditionally?

### ISSUE THREAD

**Issue by khushalsagar** [WICG/view-transitions#181]
Currently we only add the `isolation: isolate` property if the ::page-transition-container has both images. Since it's only needed to blend both and unnecessarily adding it is perf overhead. Is that ok or should we always add it.

For resolving [issue](https://tabatkins.github.io/specs/css-shared-element-transitions/#issue-703b09fb)

**Comment by jakearchibald** [comment]
It seems like a fairly straight forward UA optimisation, unless I'm missing something:

1. If the element has `isolation: isolate`:
  1. If the element contains 0 renderable children: No need to isolate.
  1. Else, if the element contain no renderable descendants with a non-normal `mix-blend-mode` (ignoring descendant isolated elements), then create a stacking context, but there's no need to apply isolation.
  1. Else, if the element contains 1 child, and none of its descendants have non-normal `mix-blend-mode` (ignoring descendant isolated elements, then create a stacking context, but there's no need to apply isolation.

**Comment by khushalsagar** [comment]
Yeah I'm going to see if we can optimize this internally. We need to detect whether isolation and mix-blend-mode are a no-op. The paint stack assumes that mix-blend-mode on any element implies that an offscreen render pass is needed, likely coz its rare that a node with a non-default mix-blend-mode is already a texture/single image. But we should be able to detect this case and optimize.

**Comment by yoavweiss** [comment]
> [!NOTE]
> 🤖 **This comment was written and posted by an AI agent** (Claude, running in the [Pi](https://github.com/badlogic/pi-mono) coding agent), acting on behalf of @yoavweiss while triaging this repository ahead of archiving it. The triage was derived from the current spec text and from open issues in `w3c/csswg-drafts`. **If it's wrong, please say so and it will be reopened** — no hard feelings, it's a bot.

**Resolved.** The conditional approach was dropped — the UA stylesheet applies `isolation: isolate` unconditionally to `::view-transition-image-pair()`. See the [UA stylesheet](https://drafts.csswg.org/css-view-transitions-1/#ua-styles); the spec notes the element "exists to provide `isolation: isolate` for its children". The `tabatkins.github.io` issue marker referenced in the OP is long gone. Closing.

---

<sub>**Where this work lives now:** View Transitions is specified in [css-view-transitions-1](https://drafts.csswg.org/css-view-transitions-1/) and [css-view-transitions-2](https://drafts.csswg.org/css-view-transitions-2/), maintained in [w3c/csswg-drafts](https://github.com/w3c/csswg-drafts/issues?q=is%3Aissue+is%3Aopen+label%3Acss-view-transitions-1%2Ccss-view-transitions-2). Please file new issues there.</sub>
